src/distance_finder.cpp

```diff
@@ -266,14 +290,41 @@ class DistanceFinder {
         double closest_angle;
         getClosestObject(selected_points, closest_distance, closest_angle);
 
+        // get all points on the circle, points that fall within a distance of each other
+        std::vector<lidarPoint> circle_points;
+        circle_points.push_back(lidarPoint(closest_distance, closest_angle));
+        for (int i = 0; i < selected_points.size(); i++) {
+            double i_dist = selected_points[i].getDistance();
+            double i_angle = selected_points[i].getAngle();
+            double dist = sqrt(pow(i_dist, 2) + pow(closest_distance, 2) - 2*i_dist*closest_distance*cos(closest_angle - i_angle));
+            if (dist < lidar_point_range_p) {
+                circle_points.push_back(selected_points[i]);
+            }
+        }
+        if (circle_points.size()<min_circle_pts_p) {
+            ROS_WARN_STREAM(TAG << "Expected at least " << min_circle_pts_p << " points for radius calculation but got " << circle_points.size());
+            return;
+        }
+
+        // do radius checking, and confirm that the prop found has the expected radius
+        lidarCalculations lidarCalc;
+        double radius = lidarCalc.calculateRadius(circle_points, min_circle_pts_p);
+        ROS_INFO_STREAM(TAG << "r = " << radius << ", close pt = " << closest_distance);
+        // TODO: change below radius validation so that we only compare against the correct prop type
+        if (!((radius > marker_radius_p - prop_range_p  && radius < marker_radius_p + prop_range_p ) || (radius > buoy_radius_p - prop_range_p && radius < buoy_radius_p + prop_range_p))) {
+            // if the prop doesn't fit a radius range, then it doesn't have a valid radius
+            // to be more specific, get label for prop then specify what radius to compare to based on that
+            ROS_WARN_STREAM(TAG << "Calculated radius does not match the expected radius");
+            return;
+        }
 
         // Publish Message
 
-        if (closest_distance < max_lidar_range_)
+        if (closest_distance < max_lidar_range_p)
```

Comment:
is `closest_distance` now actually the center of the prop since we are calculating the radius? Can we re-name that variable to `prop_center_distance` or something along those lines if so.